enhancements/api-review/custom-tags-aws.md

```diff
@@ -73,71 +95,376 @@ Note existing unchanged behavior: The installer will apply these tags to all AWS
+`.spec.platformSpec.aws` can be updated or changed which will override the values from status.
+
+For installer configuration compatibility, configuration with `experimentalPropagateUserTags` should be supported.
 
 On downgrade:
 
-The status field may remain populated, components may or may not continue to tag newly created resources w/ the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.
+The status/spec field may remain populated, components may or may not continue to tag newly created resources with the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.
 
 ### Version Skew Strategy
 
-TBD
-
 ## Implementation History
 
 ## Drawbacks
 
-If a customer decides they do not want these tags applied to new resources, there is no clean way to address that need:
+If a user decides that they do not want these tags applied to new resources, there is no clean way to address that need:
 
-1. they would have to get help from support to edit the status field
-2. they would have to manually remove the undesired tags from any existing resources
-
-In the future we will want to introduce a spec field where a customer can specify and edit these tags, which will be reflected into status when changed, and we will expect consuming components to reconcile changes to the set of tags by applying net new tags to existing resources (but they will still not remove tags that are dropped from the list of tags).
+1. User would have to get help from support to edit the status field for older version which uses `experimentalPropagateUserTags`.
+2. User would have to manually remove the undesired user-defined tags from some existing resources.
+3. User would have to manually update user-defined tags for AWS resources which are not managed by an operator.
+4. User would have to manually update user-defined tags for AWS resources where update logic is not supported by an operator.
 
 ## Alternatives
 
-TBD
+1. An operator can be designed to behave similar to how the installer's destroy behaves, namely by using the resource tagging API to search for resources with the kubernetes.io/.../cluster=owned tag.
+   The operator would reconcile on changes to the infrastructure CR to ensure that resources owned by the cluster have tags matching what is in the infrastructure CR.
+   That would get tricky when it comes to resources owned by in-cluster operators, though, particularly with tags that may have different values in the corresponding kube resources.
+   The operator would need to monitor multiple kube resources, infrastructure CR and multiple instances of different AWS resource types.
+
+2. A way to delete tags was considered when the user sets the user-defined tag value to an empty string in `.spec.platformSpec.aws.resourceTags`.
```

Comment:
What about a new property like `.spec.platformSpec.aws.resourceTagsDelete` where user enumerates tags specifically to be deleted?  If there's something set on a kube resource the deletion request is ignored for that resource.

I worry we'll get to a point somewhere in the future where we need to change how we tag resources for ROSA.  If we have to build a tool to delete tags it negates the value of this enhancement.

Reply:
@deads2k sketched out an api that would allow for deletion as:

```
spec.tags -- this is the current user desire to create.
spec.orphanTagsOnDeletion -- this is a list of tags for the platform to stop managing and not remove
status.tags -- this is the list of tags for the platform to create
status.tagsToRemove -- this is the list of tags for the platform to remove

a user adds to spec.tags and it is vetted and moved to status.tags

a user removes from spec.tags, it is compared against status.tags, and if there is no entry in orphanedTags, it is removed from status.tags and added to status.tagsToRemove.  if there is a matching entry in orphanTags, it is simply removed from status.tags
```

Reply:
@bparees is that something to be addressed in another enhancement or just dropped?  I don't see delete handled in this enhancement hence my comment.  Deletion with logic to identify what should be managed by the platform vs external tools and having the ability to transition to an external tool is what is keeping delete from being included, from what I can tell.  If we then step away from having any logic and leave it to users to explicitly state what tags need deleted then the decision on what to keep vs delete is left to the administrator.

Reply:
> @bparees is that something to be addressed in another enhancement or just dropped? 

neither, it's something that came up for discussion after this EP update had merged, and when the api changes were being reviewed.  I pasted it here so you could have some idea of how delete would work, if we were to include it as a capability and see what thoughts you'd have on it if that was the approach we took.

Reply:
Thanks.  Seems a reasonable approach, just inverse of my initial post.  But the key thing is giving the control to the user with explicit list of tags to orphan.  Sure.